examples/shader/specialized_mesh_pipeline.rs

```diff
@@ -0,0 +1,360 @@
+            vertex_attributes.push(Mesh::ATTRIBUTE_COLOR.at_shader_location(1));
+        }
+        // This will automatically generate the correct `VertexBufferLayout` based on the vertex attributes
+        let vertex_buffer_layout = layout.0.get_layout(&vertex_attributes)?;
+
+        Ok(RenderPipelineDescriptor {
+            label: Some("Specialized Mesh Pipeline".into()),
+            layout: vec![
+                // Bind group 0 is the view uniform
+                self.mesh_pipeline
+                    .get_view_layout(MeshPipelineViewLayoutKey::from(mesh_key))
+                    .clone(),
+                // Bind group 1 is the mesh uniform
+                self.mesh_pipeline.mesh_layouts.model_only.clone(),
+            ],
+            push_constant_ranges: vec![],
+            vertex: VertexState {
+                shader: self.shader_handle.clone(),
+                shader_defs: vec![],
+                entry_point: "vertex".into(),
+                // Customize how to store the meshes' vertex attributes in the vertex buffer
+                buffers: vec![vertex_buffer_layout],
+            },
+            fragment: Some(FragmentState {
+                shader: self.shader_handle.clone(),
+                shader_defs: vec![],
+                entry_point: "fragment".into(),
+                targets: vec![Some(ColorTargetState {
+                    // This isn't required, but bevy supports HDR and non-HDR rendering
+                    // so it's generally recommended to specialize the pipeline for that
+                    format: if mesh_key.contains(MeshPipelineKey::HDR) {
+                        ViewTarget::TEXTURE_FORMAT_HDR
+                    } else {
+                        TextureFormat::bevy_default()
+                    },
+                    // For this example we only use opaque meshes,
+                    // but if you wanted to use alpha blending you would need to set it here
+                    blend: None,
+                    write_mask: ColorWrites::ALL,
+                })],
+            }),
+            primitive: PrimitiveState {
+                topology: mesh_key.primitive_topology(),
+                front_face: FrontFace::Ccw,
+                cull_mode: Some(Face::Back),
+                polygon_mode: PolygonMode::Fill,
+                ..default()
+            },
+            // Note that if your view has no depth buffer this will need to be
+            // changed.
+            depth_stencil: Some(DepthStencilState {
+                format: CORE_3D_DEPTH_FORMAT,
+                depth_write_enabled: true,
+                depth_compare: CompareFunction::GreaterEqual,
+                stencil: default(),
+                bias: default(),
+            }),
+            // This is isn't required, but if you can support MSAA
+            // it's generally recommended to specialize your pipeline for it
+            multisample: MultisampleState {
```

Comment:
might be making this up pretty sure this is required and will panic if it's not set correctly?

Reply:
If you don't use MSAA it's not required. 

Well, I should update the comment a bit, setting the sample count isn't required, but you still need to use `MultisampleState::default()`